File: server/Models/Category.cs
namespace Server.Models
{
    public class Category
    {
        public int Id { get; set; }
        public string Name { get; set; } = "";
        public string IconName { get; set; } = "";

        public int? UserId { get; set; }
        public User? User { get; set; } = null!;

        public ICollection<Subcategory> Subcategories { get; set; } = [];
    }
}

File: client/src/interfaces/category.ts
export interface CategoryObject {
  id: number;
  name: string;
  iconName: string;
  userId: number;
  subcategories: Subcategory[];
}

export interface Subcategory {
  id: number;
  name: string;
  iconName: string;
  categoryId: number;
  expenses: Expense[];
}

export interface Expense {
  id: number;
  amount: number;
  amountDate: string;
  userId: number;
}

export interface CategoryAddData {
  category: string;
  iconName: string;
  subcategory: {
    subcategoryName: string;
    subcategoryIconName: string;
  }[];
}

export interface CategoryPutData {
  id: number;
  type: string;
  data: {
    name: string;
    iconName: string;
  };
}

export interface CategoryDeleteData {
  id: number;
  type: string;
}

export interface EditTarget {
  id: number;
  type: "cat" | "sub";
}
